samples/admin/snmp.py

```diff
@@ -0,0 +1,476 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module performs the operations related to snmp server, user and traps.
+"""
+
+
+def snmp_enable(handle, community=None,
+                privilege="disabled", trap_community=None,
+                sys_contact=None, sys_location=None, port="161"):
+    """
+    Enables SNMP.
+
+    Args:
+        handle (ImcHandle)
+        community (string): community
+        privilege (string): "disabled", "limited", "full"
+        trap_community(string): community to be used when generating traps
+        sys_contact (string): sys_contact
+        sys_location (string): sys_location
+        port (string): port on which SNMP agent runs
+
+    Returns:
+        CommSnmp: Managed object
+
+    Raises:
+        ImcOperationError: If CommSnmp Mo is not present
+
+    Example:
+        mo = snmp_enable(handle,
+                    community="username",
+                    sys_contact="user contact",
+                    sys_location="user location")
+    """
+
+    from imcsdk.mometa.comm.CommSnmp import CommSnmpConsts
+    from imcsdk.imcexception import ImcOperationError
+
+    dn = "sys/svc-ext/snmp-svc"
+    mo = handle.query_dn(dn)
+    if not mo:
+        raise ImcOperationError("snmp enable", "snmp config does not exist")
+
+    mo.admin_state = CommSnmpConsts.ADMIN_STATE_ENABLED
+
+    if community:
+        mo.community = community
+    if privilege:
+        mo.com2_sec = privilege
+    if trap_community:
+        mo.trap_community = trap_community
+    if sys_contact:
+        mo.sys_contact = sys_contact
+    if sys_location:
+        mo.sys_location = sys_location
+    mo.port = port
+
+    handle.set_mo(mo)
+    return mo
+
+
+def snmp_disable(handle):
+    """
+    Disables SNMP.
+
+    Args:
+        handle (ImcHandle)
+
+    Returns:
+        CommSnmp: Managed Object
+
+    Raises:
+        ValueError: If CommSnmp Mo is not present
+
+    Example:
+        snmp_disable(handle)
+    """
+
+    from imcsdk.mometa.comm.CommSnmp import CommSnmpConsts
+    from imcsdk.imcexception import ImcOperationError
+
+    dn = "sys/svc-ext/snmp-svc"
+    mo = handle.query_dn(dn)
+    if not mo:
+        raise ImcOperationError("snmp enable", "snmp config does not exist")
+
+    mo.admin_state = CommSnmpConsts.ADMIN_STATE_DISABLED
+    handle.set_mo(mo)
+    return mo
+
+
+def snmp_enabled(handle):
+    """
+    This method checks if snmp is enabled or not
+    Args:
+        handle (ImcHandle)
+
+    Returns:
+        bool
+    """
+    from imcsdk.mometa.comm.CommSnmp import CommSnmpConsts
+    from imcsdk.imcexception import ImcOperationError
+
+    dn = "sys/svc-ext/snmp-svc"
+    mo = handle.query_dn(dn)
+    if not mo:
+        raise ImcOperationError("snmp enable", "snmp config does not exist")
+
+    if mo.admin_state == CommSnmpConsts.ADMIN_STATE_ENABLED:
+        return True
+    else:
+        return False
```

Comment:
the above code can be re-written as `return mo.admin_state == CommSnmpConsts.ADMIN_STATE_ENABLED`